tests/zfs-tests/tests/functional/cli_root/zpool_create/zpool_create_024_pos.ksh

```diff
@@ -0,0 +1,145 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2016, Lawrence Livermore National Security, LLC.
+#
+
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/tests/functional/cli_root/zpool_create/zpool_create.cfg
+
+#
+# DESCRIPTION:
+# Many 'zpool create' and 'zpool destroy' must succeed concurrently.
+#
+# STRATEGY:
+# 1. Create N process each of which create/destroy a pool M times.
+# 2. Allow all process to run to completion.
+# 3. Verify all pools and their vdevs were destroyed.
+#
+
+verify_runnable "global"
+
+function cleanup
+{
+	if [[ -n "$child_pids" ]]; then
+		for wait_pid in $child_pids; do
+			$KILL $wait_pid 2>/dev/null
+		done
+	fi
+
+	if [[ -n "$child_pools" ]]; then
+		for pool in $child_pools; do
+			typeset vdev0="$TEST_BASE_DIR/$pool-vdev0.img"
+			typeset vdev1="$TEST_BASE_DIR/$pool-vdev1.img"
+
+			if poolexists $pool; then
+				destroy_pool $pool
+			fi
+
+			$RM -f $vdev0 $vdev1
+		done
+	fi
+}
+
+log_onexit cleanup
+
+log_assert "Many 'zpool create' and 'zpool destroy' must succeed concurrently."
+
+child_pids=""
+child_pools=""
+
+function zpool_stress
+{
+	typeset pool=$1
+	typeset vdev0="$TEST_BASE_DIR/$pool-vdev0.img"
+	typeset vdev1="$TEST_BASE_DIR/$pool-vdev1.img"
+	typeset -i iters=$2
+	typeset retry=10
+	typeset j=0
+
+	$TRUNCATE -s $FILESIZE $vdev0
+	$TRUNCATE -s $FILESIZE $vdev1
+
+	while [[ $j -lt $iters ]]; do
+		((j = j + 1))
+		$SLEEP 1
+
+		$ZPOOL create $pool $vdev0 $vdev1
+		if [ $? -ne 0 ]; then
+			return 1;
+		fi
+
+		typeset k=0;
+		while [[ $k -lt $retry ]]; do
```

Comment:
Good idea, I'll add a comment.  `zfs destroy` can occasionally return EBUSY when blkid is concurrently still probing the new volume.